### PR TITLE
[5.x] Switch to Node 20

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes

- Node 18 Active Support ended 2 weeks ago (18 Oct 2023), so switching to 20
- ~~Added `.npmrc` so the node version is strict~~
- Added missing `semver` package in the devDependencies
- use the helper handleErr fn
- Added a runtime conditional and a message for incompatible node versions

### Testing Instructions

Run `npm ci`, everything still works

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,8 @@
     "url": "https://github.com/joomla/joomla-cms.git"
   },
   "engines": {
-    "node": ">=18",
-    "npm": ">=9.6.7"
+    "node": "20.x",
```
Comment (C-Lodder) [inline package.json]: Please don't remove the `>=` prefix, else you're preventing STS users from performing any tasks.

Comment (dgrammatiko) [reply package.json]: Can change that. BTW what is STS?

Comment (C-Lodder) [reply package.json]: Short Term Support (All of the Node.js `odd` versions)

**Comment by bembelimen** [comment]
Hey @dgrammatiko would you please check the upmerge #42345 if this resolved this PR or if something is missing? Thanks!

**Comment by dgrammatiko** [comment]
This can be closed although there are some mistakes that need fixing. Will do a pr once I get some free time

**Pull request closed without merging**